src/main/node/config-manager.ts

```diff
@@ -1,144 +1,116 @@
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.publicationName = newName;
+  });
+}
 
-// TODO: It would be much better if the function took two args - path and an object with the options
-// TODO: The whole return of booleans is a bit weird as well, should probably throw an Error instead
-export function createConfigFile(
+export function updateDescription(path: string, newDescription: string) {
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.description = newDescription;
+  });
+}
+
+export function updateCollaborators(
   path: string,
-  name: string,
-  description: string,
-  collaborators: Collaborator[],
-  packageManager: string,
-  tag: string
-): boolean {
-  if (isDirectory(path)) {
-    const configContent = {
-      name,
-      description,
-      collaborators,
-      packageManager,
-      tag,
-    };
-    const configContentJSON = JSON.stringify(configContent, null, 2);
-    fs.writeFileSync(`${path}/${configFileName}`, configContentJSON);
-    return true;
-  }
-  return false;
+  newCollaborators: Collaborator[]
+) {
+  validateNonEmpty(newCollaborators);
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.collaborators = newCollaborators;
+  });
 }
 
-export function deleteConfigFile(path: string): boolean {
-  if (isDirectory(path) && isPublication(path)) {
-    try {
-      if (fs.existsSync(path)) {
-        fs.unlinkSync(`${path}/${configFileName}`);
-        return true;
-      }
-    } catch (err) {
-      console.error(err);
-      return false;
-    }
-  }
-  return false;
+export function updatePackageManager(path: string, newPackageManager: string) {
+  validateNonEmptyNorNull(newPackageManager);
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.packageManager = newPackageManager;
+  });
 }
 
-// TODO: Unsure how much sense it makes to have all separate `modify` functions, could probably abstract that
-export function modifyName(path: string, newName: string): boolean {
-  if (isDirectory(path) && isPublication(path)) {
-    const configContentJSON = fs.readFileSync(`${path}/${configFileName}`);
-    const configContent = JSON.parse(configContentJSON);
-    configContent.name = newName;
-    const newConfigContentJSON = JSON.stringify(configContent);
-    deleteConfigFile(path);
-    fs.writeFileSync(`${path}/${configFileName}`, newConfigContentJSON);
-    return true;
-  }
-  return false;
+export function updateTag(path: string, tag: string) {
+  const configurationToUpdate: Configuration = getConfigFile(path);
+  updateConfigField(configurationToUpdate, path, () => {
+    configurationToUpdate.tag = tag;
+  });
```

Comment:
All the update functions could be shortened to one overridable function to make everything more concise:

```ts
export function updateConfigField(
  path: string,
  field: keyof Omit<
    Configuration,
    'useTypescript' | 'useSass' | 'collaborators'
  >,
  value: string
): void;
export function updateConfigField(
  path: string,
  field: keyof Pick<Configuration, 'useTypescript' | 'useSass'>,
  value: boolean
): void;
export function updateConfigField(
  path: string,
  field: keyof Pick<Configuration, 'collaborators'>,
  value: Collaborator[]
): void;
export function updateConfigField(path: string, field: any, value: any) {
  if (field === 'publicationName') validateNonEmptyNorNull(value);

  const configuration = getConfigFile(path);
  const updatedConfiguration = { ...configuration, [field]: value };

  fs.writeFileSync(
    `${path}/${configFileName}`,
    JSON.stringify(updatedConfiguration, null, 2)
  );
}
```

With this we have one implementation with different overloads so that the typings of field and value always match and we significantly cut down on the amount of code.

It also allows to get rid of `saveUpdatedConfiguration`, `validateNonEmpty` and previous `updateConfigField` helper functions.